src/input_processing/mod.rs

```diff
@@ -0,0 +1,239 @@
+//!
+//! - [`AxisSensitivity`]: Single-axis scaling.
+//! - [`DualAxisSensitivity`]: Dual-axis scaling.
+//!
+//! ## Value Bounds
+//!
+//! Value bounds define the boundaries for constraining input values,
+//! preventing unexpected behavior that might occur outside these bounds.
+//! Values exceeding the bounds are clamped to fit within the specified range.
+//!
+//! - [`AxisBounds`]: Restricts single-axis input values to a range.
+//! - [`DualAxisBounds`]: Restricts dual-axis input values to a range along each axis.
+//! - [`CircleBounds`]: Limits dual-axis input values to a maximum magnitude.
+//!
+//! ## Deadzones
+//!
+//! ### Unscaled Versions
+//!
+//! Unscaled deadzones specify ranges where near-zero input values are excluded, treating them as zero.
+//!
+//! - [`AxisExclusion`]: Excludes single-axis input values within a specified range.
+//! - [`DualAxisExclusion`]: Excludes dual-axis input values within a specified range along each axis.
+//! - [`CircleExclusion`]: Excludes dual-axis input values below a specified magnitude threshold.
+//!
+//! ### Scaled Versions
+//!
+//! Scaled deadzones process input values by clamping them to fit within the default value bounds,
+//! considering a specified exclusion range, and scaling unchanged values linearly in between.
+//!
+//! - [`AxisDeadzone`]: Normalizes single-axis input values based on [`AxisBounds::default`] and a specified [`AxisExclusion`].
+//! - [`DualAxisDeadzone`]: Normalizes dual-axis input values based on [`DualAxisBounds::default`] and a specified [`DualAxisExclusion`].
+//! - [`CircleDeadzone`]: Normalizes dual-axis input values based on [`CircleBounds::default`] and a specified [`CircleExclusion`].
+//!
+//! ## Composite Processors
+//!
+//! Pipelines are dynamic sequence containers of processors, allowing you to create custom processing steps.
+//!
+//! - [`AxisProcessingPipeline`]: Applies a sequence of [`AxisProcessor`]s to process single-axis input values.
+//! - [`DualAxisProcessingPipeline`]: Applies a sequence of [`DualAxisProcessor`]s to process dual-axis input values.
+//!
+//! While pipelines offer flexibility in defining custom processing steps,
+//! for performance-critical scenarios, consider using the macros below for optimized implementations.
+//!
+//! - [`define_axis_processing_pipeline`](crate::define_axis_processing_pipeline):
+//!     Generates an [`AxisProcessor`] with fixed processing steps to process single-axis input values.
+//! - [`define_dual_axis_processing_pipeline`](crate::define_dual_axis_processing_pipeline):
+//!     Generates a [`DualAxisProcessor`] with fixed processing steps to process dual-axis input values.
+//!
+//! Choose between pipelines and macros based on your specific processing needs and performance requirements.
+
+pub use self::dual_axis::*;
+pub use self::single_axis::*;
+
+pub mod dual_axis;
+pub mod single_axis;
+
+// region pipeline
```

Comment:
```suggestion
```

Reply:
![image](https://github.com/Leafwing-Studios/leafwing-input-manager/assets/22796024/4abee9a6-182c-4c71-9087-27d80c74ab76)

In IDEs and vscode, `// region` can be used to collapse code sections, but now that the file length is not long, it can be removed